tests/Integration/ApiPlatform/DiscountEndpointTest.php

```diff
@@ -0,0 +1,180 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PsApiResourcesTest\Integration\ApiPlatform;
+
+use PrestaShop\PrestaShop\Core\Domain\Discount\Query\GetDiscountForEditing;
+use PrestaShop\PrestaShop\Core\Domain\Discount\Command\AddDiscountCommand;
+use PrestaShop\PrestaShop\Core\Domain\Discount\ValueObject\DiscountType;
+use Tests\Resources\Resetter\LanguageResetter;
+use Tests\Resources\Resetter\DiscountResetter;
+use Tests\Resources\ResourceResetter;
+
+class DiscountEndpointTest extends ApiTestCase
+{
+    public static function setUpBeforeClass(): void
+    {
+        parent::setUpBeforeClass();
+        (new ResourceResetter())->backupTestModules();
+        DiscountResetter::resetProducts();
+        LanguageResetter::resetLanguages();
+        self::addLanguageByLocale('fr-FR');
+        self::createApiClient(['discount_write', 'discount_read']);
+    }
+
+    public static function tearDownAfterClass(): void
+    {
+        parent::tearDownAfterClass();
+        DiscountResetter::resetProducts();
+        LanguageResetter::resetLanguages();
+        // Reset modules folder that are removed with the FR language
+        (new ResourceResetter())->resetTestModules();
+    }
+
+    /**
+     * @depends testAddDiscount
```

Comment:
Good! you used the `@depends` so `testAddDiscount` will be executed before in any case, but to ease the readability of this test class can you move this one after `testAddDiscount` please?